spec.bs

```diff
@@ -538,45 +538,161 @@ A <dfn export for=fencedframetype>size</dfn> is a struct with non-negative integ
 "size">width</dfn> and non-negative integer <dfn export for="size">height</dfn>. <span class=XXX>
 TODO: Maybe change the numeric type.</span>
 
-An <dfn export for=fencedframetype>interest group descriptor</dfn> is a struct with <dfn export
-for="interest group descriptor">owner</dfn>, which is a string, and <dfn export for="interest group
-descriptor">name</dfn>, which is a string.
+An <dfn export for=fencedframetype>interest group descriptor</dfn> is a struct with the following
+[=struct/items=]:
+
+<dl export dfn-for="interest group descriptor">
+  : <dfn>owner</dfn>
+  :: a [=string=]
+
+  : <dfn>name</dfn>
+  :: a [=string=]
+</dl>
 
 An <dfn export for=fencedframetype>exhaustive set of sandbox flags</dfn> is a [=sandboxing flag
 set=].
 
-A <dfn export for=fencedframetype>fenced frame reporter</dfn> is <span class=XXX>TODO: Specify the type for this.</span>
+An <dfn export for=fencedframetype>exhaustive set of permissions</dfn> is a [=list=] of
+[=policy-controlled features=].
+
+A <dfn export for=fencedframetype>pending event</dfn> is a struct with the following [=struct/items=]:
+<dl export dfn-for="pending event">
+  : <dfn>destination</dfn>
+  :: a {{FenceReportingDestination}}
+
+  : <dfn>eventType</dfn>
+  :: a [=string=]
+
+  : <dfn>eventData</dfn>
+  :: a [=string=]
+</dl>
+
+A <dfn export for=fencedframetype>fenced frame reporting map</dfn> is a [=map=] whose [=map/keys=]
+are {{FenceReportingDestination}}s and whose [=map/values=] are either [=lists=] of
+[=fencedframetype/pending events=] (which are used to represent events that need to be reported
+asynchronously, because the metadata has not been finalized yet), or [=maps=] whose [=map/keys=]
+are [=strings=] and whose [=map/values=] are [=urls=] (which are used to represent the actual
+metadata once it is finalized).
 
 <div algorithm>
-  In order to <dfn export>report an event</dfn>, run these steps:
+  In order to <dfn export>finalize a reporting destination</dfn>, given a [=fencedframetype/fenced
```

Comment:
Yeah, this is how FLEDGE can hook in asynchronously. Tomorrow I'll add something like:

```
Note: By keeping around a reference to the fenced frame reporting map, config-generating APIs can use this algorithm to resolve the values of reporting destinations asynchronously, after they've already constructed and returned the fenced frame config (and even after the config has been loaded). Any reports sent to this pending reporting destination will be delayed until the destination is finalized.
```

Reply:
Oh, it sounds like you wanted more clarification on which "threads" do what; I'll clarify "Any reports sent to this pending reporting destination will be delayed until the destination is finalized."

Reply:
It sounds like it will be finalized from in parallel I imagine? That note seems good, and yeah maybe just clarifying where the finalization will happen would be good.

Reply:
Added an expanded note.